test/lib/ufe/testDeleteCmd.py

```diff
@@ -526,6 +526,82 @@ def testDeleteAndRemoveConnections(self):
         self.assertFalse(surface2Prim.HasProperty('outputs:out'))
         self.assertFalse(surface3Prim.HasProperty('inputs:bsdf'))
 
+    def testDeleteAndRemoveReference(self):
+        '''Test deleting a prim and its references'''
+
+        # open treeRef.ma scene in testSamples
+        mayaUtils.openTreeRefScene()
+
+        # validate the default edit target to be the Rootlayer.
+        mayaPathSegment = mayaUtils.createUfePathSegment('|TreeRef_usd|TreeRef_usdShape')
+        stage = mayaUsd.ufe.getStage(str(mayaPathSegment))
+        self.assertTrue(stage)
+        self.assertEqual(stage.GetEditTarget().GetLayer(), stage.GetRootLayer())
+        self.assertTrue(stage.GetPrimAtPath('/TreeBase/leaf_ref_1').HasAuthoredReferences())
+        
+        # delete the greenLeaf and check the internal references were cleaned:
+        cmds.delete('|TreeRef_usd|TreeRef_usdShape,/TreeBase/leavesXform/greenLeaf')
+        self.assertFalse(stage.GetPrimAtPath('/TreeBase/leaf_ref_1').HasAuthoredReferences())
+        # Ref2 has an "external" reference, but to this file. It will not be
+        # cleaned up because external, baut maybe it should have since the
+        # referenced file happens to be the root layer?
+        self.assertTrue(stage.GetPrimAtPath('/TreeBase/leaf_ref_2').HasAuthoredReferences())
```

Comment:
Indeed, might be good to file a story to improve this, along with my comments about other list ops.

Reply:
Also small typo baut -> but, no need to respin teh PR just for that in a unit test though